controllers/manilascheduler_controller.go

```diff
@@ -558,46 +493,100 @@ func (r *ManilaSchedulerReconciler) reconcileUpgrade(ctx context.Context, instan
 	return ctrl.Result{}, nil
 }
 
-// generateServiceConfigMaps - create custom configmap to hold service-specific config
+// getSecret - get the specified secret, and add its hash to envVars
+func (r *ManilaSchedulerReconciler) getSecret(
```

Comment:
just out of curiosity: is there a way we can reuse this function? Might be another newbie question too :)
Asking because I've realized the only thing changing is *manilav1beta1.ManilaAPI -> *manilav1beta1.ManilaScheduler

Reply:
so I'm planning to have a follow up on that front, however, I think I have two options there:

1. Use generics and let the function not being tied to any reconciler
2. Create a `BaseReconciler` and let `Manila{API, Scheduler, Share}` do inheritance from it

I see option 2 more realistic and cleaner rather than 1, but I have to experiment with it a bit more, hence targeting for a future improvement when the goal will be doing code cleanup and double check it's stable (mostly Q1 next year).

Reply:
Yeah, I think option 2 is a better approach too. Thanks!